### DISCUSSION TITLE
Mockoon as GitLab CI service

### DISCUSSION THREAD

**Discussion by platkouski** [mockoon/mockoon#1160] [Help]
Hi, could you give me a hand, please. How can I integrate Mockoon to the GitLab CI pipeline as a service?

I tried the following:
```
  services:
    - name: mockoon/cli:5
      alias: mockoon
      command: [ "--data", "~/src/test/resources/qwe.json" ]
```
But have this issue:
```
Error: This file is not a valid OpenAPI specification (JSON or YAML v2.0.0
and v3.0.0) or Mockoon environment: ENOENT: no such file or directory, 
open '~/src/test/resources/qwe.json'
```

As I understand the main issue is 'no such file or directory', but my file is already exist in repository. Absolute path is not working as partial too.

**Comment by 255kb** [comment]
Hi, not a GitLab specialist, but shouldn't the path be relative instead? `./src/test/resources/qwe.json` or `src/test/resources/qwe.json`

**Reply by platkouski** [reply]
Hi @255kb, thank you for the response!
Yes, I tried, but there is the same issue. 😢 
```
ENOENT: no such file or directory, open './src/test/resources/qwe.json'
ENOENT: no such file or directory, open '/src/test/resources/qwe.json'
ENOENT: no such file or directory, open 'src/test/resources/qwe.json'
```
Full path is `/project/src/test/resources/qwe.json` and I'm located in `project` directory.

**Reply by 255kb** [reply]
Could it be that because you are using a service, you are not in the directory where the repository is cloned (`/builds/$CI_PROJECT_PATH`)? You may have to `cd` to this directory first and maybe even wait for it to be present according to this documentation: https://docs.gitlab.com/ee/ci/services/#shared-builds-directory

**Reply by platkouski** [reply]
Yes, looks like that service is starting before initializing git repository where my file is located. I will try to find a workaround, thank you!
